cookbook/core/containerization/image_sepc_example.py

```diff
@@ -0,0 +1,117 @@
+"""
+.. _image_spec_example:
+
+Building Image without Dockerfile
+---------------------------------
+
+.. tags:: Containerization, Intermediate
+
+Image Spec is a way to specify how to build a container image without a Dockerfile. The image spec by default will be
+converted to an `Envd <https://envd.tensorchord.ai/>`__ config, and the `Envd builder
+<https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-envd/flytekitplugins/envd
+/image_builder.py#L12-L34>`__ will build the image for you. However, you can also register your own builder to build
+the image using other tools.
+
+For every :py:class:`flytekit.PythonFunctionTask` task or a task decorated with the ``@task`` decorator,
+you can specify rules for binding container images. By default, flytekit binds a single container image, i.e.,
+the `default Docker image <https://ghcr.io/flyteorg/flytekit>`__, to all tasks. To modify this behavior,
+use the ``container_image`` parameter available in the :py:func:`flytekit.task` decorator, and pass an
+`ImageSpec`.
+
+Before building the image, Flytekit checks the container registry first to see if the image already exists. By doing
+so, it avoids having to rebuild the image over and over again. If the image does not exist, flytekit will build the
+image before registering the workflow, and replace the image name in the task template with the newly built image name.
+
+"""
+# %%
+# .. admonition:: Prerequisites
+#    :class: important
+#
+#    - Install `flytekitplugins-envd <https://github.com/flyteorg/flytekit/tree/master/plugins/flytekit-envd>`__ to build the image spec.
+#    - To build the image on remote machine, check this `doc <https://envd.tensorchord.ai/teams/context.html#start-remote-buildkitd-on-builder-machine>`__
+#
+#
+import pandas as pd
+from flytekit import ImageSpec, task, workflow, Resources
+
+# %%
+# Image Spec
+# ==========
+#
+# People can specify python packages, apt packages, and environment variables. Those packages will be added on top of
+# the `default image <https://github.com/flyteorg/flytekit/blob/master/Dockerfile>`__. You can also override the
+# default image by passing ``base_image`` parameter to the ``ImageSpec``.
+pandas_image_spec = ImageSpec(
+    base_image="ghcr.io/flyteorg/flytekit:py3.8-1.6.0",
+    packages=["pandas", "numpy"],
+    python_version="3.9",
+    apt_packages=["git"],
+    env={"Debug": "True"},
+)
+
+sklearn_image_spec = ImageSpec(
+    base_image="ghcr.io/flyteorg/flytekit:py3.8-1.6.0",
+    packages=["tensorflow"],
+)
+
+# %%
+# ``is_container`` is used to check if the task is using the image built from the image spec.
+# If the task is using the image built from the image spec, then the task will import tensorflow.
+# It can reduce module loading time and avoid unnecessary dependency installation in the single image.
```

Comment:
```suggestion
# ``is_container`` is used to determine whether the task is utilizing the image constructed from the ``ImageSpec``.
# If the task is indeed using the image built from the ``ImageSpec``, it will then import Tensorflow.
# This approach helps minimize module loading time and prevents unnecessary dependency installation within a single image.
```